src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -86,12 +114,14 @@ object IntervalTree {
       var pruned = 0
       while (i < unpruned.length) {
         val interval = unpruned(i)
-        if (pord.lteq(interval.start, tmp.end)) {
-          val max = if (pord.gt(interval.end, tmp.end))
-            interval.end
+        val c = pord.compare(interval.start, tmp.end)
+        if (c < 0 || (c == 0 && (interval.includeStart || tmp.includeEnd))) {
+          tmp = if (pord.lt(interval.end, tmp.end))
+            Interval(tmp.start, tmp.end, tmp.includeStart, tmp.includeEnd)
+          else if (pord.eq(interval.end, tmp.end))
+            Interval(tmp.start, tmp.end, tmp.includeStart, tmp.includeEnd || interval.includeEnd)
           else
-            tmp.end
-          tmp = Interval(tmp.start, max)
+            Interval(tmp.start, interval.end, tmp.includeStart, interval.includeEnd)
```

Comment:
Will this update the endpoints properly if I have
`pruned = (1, 5), [1, 5], (10, 20), [10, 20]...`

I'd expect the result to be `[1,5], [10, 20]`.